src/shogun/preprocessor/RandomFourierGaussPreproc.cpp

```diff
@@ -140,101 +123,66 @@ EFeatureType RandomFourierGaussPreproc::get_feature_type() {
-	return ((int32_t) dim_input_space);
+	return m_randomweights.num_cols;
 }
 
-void RandomFourierGaussPreproc::set_kernelwidth(const float64_t kernelwidth2 ) {
-	if (kernelwidth2 <= 0) {
-		throw ShogunException(
-				"void CRandomFourierGaussPreproc::set_kernelwidth(const float64_t kernelwidth2 ): kernelwidth2 <= 0 is not allowed");
+void RandomFourierGaussPreproc::set_kernelwidth(const float64_t kernelwidth ) {
+	if (kernelwidth <= 0) {
+		error("InputError: kernelwidth must be greater 0!");
 	}
-	kernelwidth=kernelwidth2;
-}
 
-float64_t RandomFourierGaussPreproc::get_kernelwidth( ) const {
-	return (kernelwidth);
+	is_updated = m_kernelwidth == kernelwidth;
+	SG_DEBUG("set_kernelwidth(), is_updated = {}",is_updated);
+	m_kernelwidth=kernelwidth;
 }
 
-void RandomFourierGaussPreproc::set_dim_input_space(const int32_t dim) {
-	if (dim <= 0) {
-		throw ShogunException(
-				"void CRandomFourierGaussPreproc::set_dim_input_space(const int32 dim): dim<=0 is not allowed");
-	}
-
-	dim_input_space = dim;
-
-}
-
-bool RandomFourierGaussPreproc::test_rfinited() const {
-
-	if ((dim_feature_space ==  cur_dim_feature_space)
-			&& (dim_input_space > 0) && (dim_feature_space > 0)) {
-		if ((dim_input_space == cur_dim_input_space)&&(Math::abs(kernelwidth-cur_kernelwidth)<1e-5)) {
-
-			// already inited
-			return true;
-		} else {
-			return false;
-		}
-	}
-
-	return false;
+float64_t RandomFourierGaussPreproc::get_kernelwidth( ) const {
+	return (m_kernelwidth);
 }
 
-bool RandomFourierGaussPreproc::init_randomcoefficients() {
-	if (dim_feature_space <= 0) {
-		throw ShogunException(
-				"bool CRandomFourierGaussPreproc::init_randomcoefficients(): dim_feature_space<=0 is not allowed\n");
+bool RandomFourierGaussPreproc::init_randomcoefficients(int32_t dim_input_space) {
+	if (m_dim_feature_space <= 0) {
+		error("ValueError: m_dim_feature_space has value <=0 which isn't allowed!, please set valid value via corresponding setter\n");
 	}
 	if (dim_input_space <= 0) {
-		throw ShogunException(
-				"bool CRandomFourierGaussPreproc::init_randomcoefficients(): dim_input_space<=0 is not allowed\n");
+		error("InputError: dim_input_space must be greater 0!\n");
 	}
 
-	if (test_rfinited()) {
+	if (is_updated)
 		return false;
-	}
-
-
-	io::info("initializing randomcoefficients ");
-
-	float64_t pi = 3.14159265;
 
+	io::info("initializing randcoefficients ");
 
-	SG_FREE(randomcoeff_multiplicative);
-	randomcoeff_multiplicative=NULL;
-	SG_FREE(randomcoeff_additive);
-	randomcoeff_additive=NULL;
-
-
-	cur_dim_feature_space=dim_feature_space;
-	randomcoeff_additive=SG_MALLOC(float64_t, cur_dim_feature_space);
-	cur_dim_input_space = dim_input_space;
-	randomcoeff_multiplicative=SG_MALLOC(float64_t, cur_dim_feature_space*cur_dim_input_space);
-
-	cur_kernelwidth=kernelwidth;
+	SG_FREE(m_randomweights);
+	m_randomweights=NULL;
+	SG_FREE(m_randomoffset);
+	m_randomoffset=NULL;
 
+	m_randomoffset= std::make_shared<SGVector<float64_t>>()
```

Comment:
We don't do this. We always create SGVector on the stack and pass it by value or reference